### ISSUE TITLE
Segmentation fault with WARP

### ISSUE THREAD

**Issue by FlorianWilhelm** [lyst/lightfm#117]
Using the WARP model gives me right now a segmentation fault under 64bit Linux using an environment with conda (4.0.5), conda-env (2.4.5), lightfm (1.9), numpy (1.11.2), pandas (0.18.0), scipy (0.18.1). The attached script outputs:
```
Shape: (125288, 792409)
NNZ: 5013789
Segmentation fault (core dumped)
```
Reducing the shape to (100000, 10000) works. Also changing the model to logistic works too. Therefore I would say there is a strong indication that it is related to the WARP cython code itself.
My laptop has 32GB RAM so memory size should not be a problem of this segmentation fault. 
```python
import numpy as np
import scipy as sp
from scipy import stats
import pandas as pd
from lightfm import LightFM
from lightfm.evaluation import precision_at_k

from coo_matrix import *

M = load_coo('./intrct_mat.npz')

# Uncommenting these makes it work
#M = M.tocsr()
#M = M[:100000,:10000].tocoo()

print('Shape: {}'.format(M.shape))
print('NNZ: {}'.format(M.nnz))
model = LightFM(loss='warp')  # using 'logistic' here makes it work
model.fit(M, epochs=30, num_threads=1)
```

**Comment by maciejkula** [comment]
Thanks for the report. I'll attempt to reproduce and find a fix.

**Comment by maciejkula** [comment]
The fix will be pushed to PyPI in the next release of LightFM.

**Comment by maciejkula** [comment]
Fix included in the 1.10 release, available on PyPI.
